options/db_options.cc

```diff
@@ -332,6 +333,30 @@ std::unordered_map<std::string, OptionTypeInfo>
          {offsetof(struct DBOptions, best_efforts_recovery),
           OptionType::kBoolean, OptionVerificationType::kNormal,
           OptionTypeFlags::kNone, 0}},
+        // The following properties were handled as special cases in ParseOption
+        // This means that the properties could be read from the options file
+        // but never written to the file or compared to each other.
+        {"rate_limiter_bytes_per_sec",
+         {offsetof(struct DBOptions, rate_limiter), OptionType::kUnknown,
+          OptionVerificationType::kNormal,
+          (OptionTypeFlags::kStringNone | OptionTypeFlags::kCompareNever), 0,
+          [](const ConfigOptions& /*opts*/, const std::string& /*name*/,
+             const std::string& value, char* addr) {
+            auto limiter =
+                reinterpret_cast<std::shared_ptr<RateLimiter>*>(addr);
+            limiter->reset(NewGenericRateLimiter(
+                static_cast<int64_t>(ParseUint64(value))));
+            return Status::OK();
+          }}},
+        {"env",
+         {offsetof(struct DBOptions, env), OptionType::kUnknown,
+          OptionVerificationType::kNormal,
+          (OptionTypeFlags::kStringNone | OptionTypeFlags::kCompareNever), 0,
+          [](const ConfigOptions& /*opts*/, const std::string& /*name*/,
+             const std::string& value, char* addr) {
+            auto env = reinterpret_cast<Env**>(addr);
+            return Env::LoadEnv(value, env);
```

Comment:
Although in LoadEnv, if ObjectRegistry::NewInstance()->NewStaticObject<Env>(value, &env) fails, env will not be changed. To be safe (e.g., LoadEnv might be changed in the future and does not provide such protection), it would be better to follow the original logic.

```
    Env* env = new_options->env;
      Status status = Env::LoadEnv(value, &env);
      // Only support static env for now.
      if (status.ok()) {
        new_options->env = env;
      }
```